src/bsoncxx/lib/bsoncxx/v1/decimal128.cpp

```diff
@@ -16,13 +16,117 @@
 
 //
 
+#include <bsoncxx/v1/exception.hpp>
+
+#include <climits>
+#include <cstddef>
+#include <string>
+#include <system_error>
+
+#include <bsoncxx/private/bson.hh>
+#include <bsoncxx/private/immortal.hh>
 #include <bsoncxx/private/type_traits.hh>
 
 namespace bsoncxx {
 namespace v1 {
 
+using code = v1::decimal128::errc;
+
 static_assert(is_regular<decimal128>::value, "bsoncxx::v1::decimal128 must be regular");
 static_assert(is_semitrivial<decimal128>::value, "bsoncxx::v1::decimal128 must be semitrivial");
 
+decimal128::decimal128(v1::stdx::string_view sv) {
+    if (sv.empty()) {
+        throw v1::exception{code::empty_string};
+    }
+
+    if (sv.size() > std::size_t{INT_MAX}) {
+        throw v1::exception{code::invalid_string_length};
+    }
```

Comment:
These checks are [not present in v_noabi](https://github.com/mongodb/mongo-cxx-driver/blob/ad622035ec9c9ab352b42449d5fcbf426e8782e9/src/bsoncxx/lib/bsoncxx/v_noabi/bsoncxx/decimal128.cpp#L26).